crates/web5/src/test_vectors.rs

```diff
@@ -207,4 +208,100 @@ mod test_vectors {
             }
         }
     }
+
+    mod crypto_ed25519 {
+        use super::*;
+        use crate::crypto::dsa::ed25519::{Ed25519Signer, Ed25519Verifier};
+        use crate::crypto::dsa::{Signer, Verifier};
+        use crate::crypto::jwk::Jwk;
+
+        #[derive(Debug, serde::Deserialize)]
+        struct SignVectorInput {
+            data: String,
+            key: Jwk,
+        }
+
+        #[derive(Debug, serde::Deserialize)]
+        struct VerifyVectorInput {
+            data: String,
+            key: Jwk,
+            signature: String,
+        }
+
+        #[test]
+        fn sign() {
+            let path = "crypto_ed25519/sign.json";
+            let vectors: TestVectorFile<SignVectorInput, Option<String>> =
+                TestVectorFile::load_from_path(path);
+
+            for vector in vectors.vectors {
+                let input = vector.input;
+                let expected_output = vector.output;
+
+                let signer = Ed25519Signer::new(input.key);
+
+                let data = hex_string_to_byte_array(input.data.to_string());
+                let result = signer.sign(&data);
+
+                if matches!(vector.errors, Some(true)) {
+                    assert!(result.is_err(), "Expected an error, but signing succeeded");
+                } else {
+                    let signature = result.expect("Signing should not fail");
+
+                    // Convert the signature to a hex string
+                    let signature_hex = byte_array_to_hex_string(&signature);
```

Comment:
I'm surprised this works. ed25519 signatures are different every time because they use a random nonce to prevent computing the private key if the same private key is used for multiple signatures

Reply:
oh no.. 
This works in every sdk

Reply:
I think we are good - https://cryptobook.nakov.com/digital-signatures/eddsa-and-ed25519

Reply:
see [Section 4.2.5 on Deterministic Nonce Generation](https://eprint.iacr.org/2020/823.pdf)

> Ed25519 uses deterministic signing which removes the need for fresh random numbers during the signing process. This does not lead to any particular consequences for our security analysis since we model the key derivation function as a random oracle. However, it is well known not to reduce security